### PR TITLE
 refactor sentinel gateway module

### PR DESCRIPTION
**Pull request by fangjian0423**
The configurations for sentinel zuul:

```properties
spring.cloud.sentinel.zuul.order.pre=...
spring.cloud.sentinel.zuul.order.post=...
spring.cloud.sentinel.zuul.order.error=...
```

The configurations for sentinel spring-cloud-gateway:

response mode: 
```properties
spring.cloud.sentinel.scg.fallback.mode=response
spring.cloud.sentinel.scg.fallback.response-status=...
spring.cloud.sentinel.scg.fallback.response-body=...
```

redirect mode: 
```properties
spring.cloud.sentinel.scg.fallback.mode=redirect
spring.cloud.sentinel.scg.fallback.redirect=...
```

### CODE REVIEW

**Comment by fangjian0423** [comment]
@sczyh30 please review it.

File: spring-cloud-alibaba-sentinel-gateway/src/main/java/org/springframework/cloud/alibaba/sentinel/gateway/scg/SentinelSCGAutoConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+	public SentinelSCGAutoConfiguration(
+			ObjectProvider<List<ViewResolver>> viewResolversProvider,
+			ServerCodecConfigurer serverCodecConfigurer) {
+		this.viewResolvers = viewResolversProvider.getIfAvailable(Collections::emptyList);
+		this.serverCodecConfigurer = serverCodecConfigurer;
+	}
+
+	private void initAppType() {
+		System.setProperty(SentinelConfig.APP_TYPE,
```
Comment (fangjian0423) [inline spring-cloud-alibaba-sentinel-gateway/src/main/java/org/springframework/cloud/alibaba/sentinel/gateway/scg/SentinelSCGAutoConfiguration.java]: Sentinel Spring Cloud Gateway APP_TYPE = 1

File: spring-cloud-alibaba-sentinel-gateway/src/main/java/org/springframework/cloud/alibaba/sentinel/gateway/zuul/SentinelZuulAutoConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	private Optional<RequestOriginParser> requestOriginParserOptional;
+
+	@Autowired
+	private SentinelZuulProperties zuulProperties;
+
+	@PostConstruct
+	private void init() {
+		requestOriginParserOptional
+				.ifPresent(ZuulGatewayCallbackManager::setOriginParser);
+		System.setProperty(SentinelConfig.APP_TYPE,
```
Comment (fangjian0423) [inline spring-cloud-alibaba-sentinel-gateway/src/main/java/org/springframework/cloud/alibaba/sentinel/gateway/zuul/SentinelZuulAutoConfiguration.java]: Sentinel Zuul APP_TYPE = 12

File: spring-cloud-alibaba-sentinel-gateway/src/main/java/org/springframework/cloud/alibaba/sentinel/gateway/scg/SentinelSCGAutoConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+		}
+		if (ConfigConstants.FALLBACK_MSG_RESPONSE.equals(fallbackProperties.getMode())) {
+			if (StringUtil.isNotBlank(fallbackProperties.getResponseBody())) {
+				GatewayCallbackManager.setBlockHandler(new BlockRequestHandler() {
+					@Override
+					public Mono<ServerResponse> handleRequest(ServerWebExchange exchange,
+							Throwable t) {
+						return ServerResponse
+								.status(fallbackProperties.getResponseStatus())
+								.contentType(MediaType.APPLICATION_JSON_UTF8)
```
Comment (sczyh30) [inline spring-cloud-alibaba-sentinel-gateway/src/main/java/org/springframework/cloud/alibaba/sentinel/gateway/scg/SentinelSCGAutoConfiguration.java]: The response content type might not be JSON?

Comment (fangjian0423) [reply spring-cloud-alibaba-sentinel-gateway/src/main/java/org/springframework/cloud/alibaba/sentinel/gateway/scg/SentinelSCGAutoConfiguration.java]: yes, i think we can add a new configuration item `spring.cloud.sentinel.scg.fallback.content-type`, default value is `application/json;charset=UTF-8`.

How do you think about it?

Comment (sczyh30) [reply spring-cloud-alibaba-sentinel-gateway/src/main/java/org/springframework/cloud/alibaba/sentinel/gateway/scg/SentinelSCGAutoConfiguration.java]: +1 for this

**Comment by fangjian0423** [comment]
👍

**Pull request merged**